feat: allow users to @People in DingTalk notifications

boomboomxx opened the pull request:
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [√] I have read and understand the pull request rules.

# Description
Sometimes we need to notify specified users in a webhook instead of all users in Dingtalk, and right now it only supports `@everyone`.
According to [Doc](https://open.dingtalk.com/document/orgapp/custom-bot-send-message-type), we can tell that it supports that feature.
Therefore, this PR adds the feature and adds it to the test message. It supports two ways of specifying:
1. mobile list
2. user ID list



## Type of change

Please delete any options that are not relevant.

- User interface (UI)
- Other

## Checklist

- [√] My code follows the style guidelines of this project
- [√] I ran ESLint and other linters for modified files
- [√] I have performed a self-review of my own code and tested it
- [√] I have commented my code, particularly in hard-to-understand areas (including JSDoc for methods)
- [√] My changes generates no new warnings
- [√] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)
![image](https://github.com/user-attachments/assets/bbb05907-60e3-402c-a3b1-c815f5a0a347)
![image](https://github.com/user-attachments/assets/2addcf6f-4753-4503-a913-3be83ffa3d19)

CommanderStorm commented on src/components/notifications/DingDing.vue:
```diff
@@ -19,8 +19,18 @@
+            <option value="specify-users">{{ $t("Mention User List") }}</option>
         </select>
     </div>
+    <div v-if="$parent.notification.mentioning === 'specify-mobiles'" class="mb-3">
```
Could we switch over to `vue-multiselect` for these instead?
I think the UX would be much nicer and less confusing.

What do you think?

boomboomxx replied:
Yes, it's better for user use, i'll change it

CommanderStorm commented on server/notification-providers/dingding.js:
```diff
@@ -11,17 +11,22 @@ class DingDing extends NotificationProvider {
+        const mentionAll = notification.mentioning === "everyone";
+        const mobileList = notification.mentioning === "specify-mobiles" ? notification.mobileList.split(",") : [];
+        const userList = notification.mentioning === "specify-users" ? notification.userList.split(",") : [];
+        const mentionStr = [ ...mobileList || [], ...userList || [] ].map(item => `@${item}`).join(" ");
```
please handle users
- using spaces => trimming the input
- using `@` => trimming the input unless this is allowed
- (optional) drop duplicates from the list

boomboomxx replied:
Thx for review, i'll change it.

CommanderStorm commented on server/notification-providers/dingding.js:
```diff
@@ -11,17 +11,22 @@ class DingDing extends NotificationProvider {
                     markdown: {
                         title: `[${this.statusToString(heartbeatJSON["status"])}] ${monitorJSON["name"]}`,
-                        text: `## [${this.statusToString(heartbeatJSON["status"])}] ${monitorJSON["name"]} \n> ${heartbeatJSON["msg"]}\n> Time (${heartbeatJSON["timezone"]}): ${heartbeatJSON["localDateTime"]}`,
+                        text: `## [${this.statusToString(heartbeatJSON["status"])}] ${monitorJSON["name"]} \n> ${heartbeatJSON["msg"]}\n> Time (${heartbeatJSON["timezone"]}): ${heartbeatJSON["localDateTime"]}${"\n\n" + mentionStr}`,
```
If we notify users below (or does `atUserIds`/.. work differently), does adding this to the notification still make sense?

boomboomxx replied:
Yes, the documentation clearly states that you need to add `@user` to the text field.

![image](https://github.com/user-attachments/assets/61623a15-a36a-422d-94c1-a59c67c55bc6)

CommanderStorm replied:
Seems weird, but ok

CommanderStorm reviewed (changes requested):
Overall, looks like a reasonable change.
I have left a few changes that I would like to see

boomboomxx commented:
> Overall, looks like a reasonable change. I have left a few changes that I would like to see
I have a question, how do I check the required rules in `vue-multiselect`.
Do you have any ideas?

CommanderStorm commented:
required rules?

boomboomxx commented:
> required rules?

Like `<input required>`, the `vue-multiselect` dose not have `required` attr.

boomboomxx commented:
Please help to review again.
I changed the `input` to `vue-multiselect` and changed the data validation rules to do the data is empty compatibility.
Currently there is only one problem, how to check if the value of `vue-multiselect` is null on submit time

CommanderStorm reviewed (approved):
Thanks, LGTM now
